contracts/Registrar.sol

```diff
@@ -1,24 +1,22 @@
 // SPDX-License-Identifier: GPL-3.0-only
+// solhint-disable no-empty-blocks
 pragma solidity ^0.7.5;
 
 import "@ensdomains/ens/contracts/ENS.sol";
-
-import "./PriceOracle.sol";
-import "./Exchange.sol";
-import "./Rad.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20Burnable.sol";
 
 contract Registrar {
     /// The ENS registry.
     ENS public immutable ens;
 
     /// The price oracle.
-    PriceOracle public oracle;
+    address public oracleAddress;
 
     /// The Rad/Eth exchange.
-    Exchange public exchange;
+    address public exchangeAddress;
 
     /// The Rad ERC20 token.
-    Rad public immutable rad;
+    address public immutable radAddress;
```

Comment:
I think that it can be of type `Rad`, AFAIK the contract type is just an address, so they'll compile to the same ABI

Reply:
Oh so internally, the storage layout is the same? That makes sense..

Reply:
I've double checked it and it's correct, contracts are just addresses: https://docs.soliditylang.org/en/v0.7.5/abi-spec.html#mapping-solidity-to-abi-types